test/test_erb_processor.rb

```diff
@@ -27,6 +27,27 @@ def tset_compile_js_erb_template
     assert_equal output, Sprockets::ERBProcessor.call(input)[:data]
   end
 
+  def test_compile_erb_template_that_depends_on_env
+    root = File.expand_path("../fixtures", __FILE__)
+    environment = Sprockets::Environment.new(root)
+    environment.append_path 'default'
+
+    ::ENV['ERB_ENV_TEST_VALUE'] = 'success'
```

Comment:
We should clean this up at the end of the test (set it back to nil).

And let's also throw in proof that an un-set variable still gets counted as a dependency.